Lib/functools.py

```diff
@@ -928,27 +928,41 @@ class singledispatchmethod:
     """
 
     def __init__(self, func):
+        import weakref # lazy import
```

Comment:
This comment is a bit fatuous at the moment — it's obvious that it's lazy; the interesting thing is _why_ we're using a lazy import here. I'd either change the comment so that it says `see comment in singledispatch for why we import weakref lazily`, or just copy that comment out in full (it's not that long)